### PR TITLE
websocket - clientside reconnect logic

### PR DESCRIPTION
**Pull request by ly5156**
https://github.com/rancher/longhorn/issues/145

### CODE REVIEW

File: src/utils/websocket.js [hunk 1]
```diff
@@ -13,11 +15,15 @@ function constructWebsocketURL(type, period) {
 
 export function wsChanges(dispatch, type, period) {
   let url = constructWebsocketURL(type, period)
-  const ws = new WebSocket(url)
-  ws.onmessage = function (msg) {
+  const options = {
+    connectionTimeout: 10000,
+    maxRetries: 10,
```
Comment (LLParse) [inline src/utils/websocket.js]: Can we remove this line? The default is `Infinity` with geometric backoff, which is what we want.

This is my test:
1. Null route websocket traffic (however you want to do it, deleting manger is fine too)
2. Refresh UI page
3. Wait 101 seconds
4. Restore websocket connectivity

Comment (yasker) [reply src/utils/websocket.js]: Yeah, that sounds good.

@LLParse can you fork this PR and update it? Remember give credit to @ly5156 .

File: src/utils/websocket.js [hunk 2]
```diff
@@ -13,11 +15,15 @@ function constructWebsocketURL(type, period) {
 
 export function wsChanges(dispatch, type, period) {
   let url = constructWebsocketURL(type, period)
-  const ws = new WebSocket(url)
-  ws.onmessage = function (msg) {
+  const options = {
+    connectionTimeout: 10000,
```
Comment (LLParse) [inline src/utils/websocket.js]: The default of `4000` is more responsive. I've heard the figure `500ms` quoted as a reasonable inter-continental RTT. There could be other high latency situations that I'm not considering.

**Comment by yasker** [comment]
Close this in favor of https://github.com/rancher/longhorn-ui/pull/69

**Pull request closed without merging**